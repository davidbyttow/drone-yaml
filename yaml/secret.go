package yaml

import "errors"

type (
	// Secret is a resource that provides encrypted data
	// and pointers to external data (i.e. from vault).
	Secret struct {
		Kind string `json:"kind,omitempty"`
		Type string `json:"type,omitempty"`

		Data map[string]string `json:"data,omitempty"`
	}
)

// Validate returns an error if the secret is invalid.
func (s *Secret) Validate() error {
	if len(s.Data) == 0 {
		return errors.New("yaml: invalid secret resource")
	}
	return nil
}
