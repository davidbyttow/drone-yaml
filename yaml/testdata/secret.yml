---
kind: secret
type: encrypted

data:
  username: N2NmYjA3ODQwNTY1ODFlY2E5MGJmOWI1NDk0NDFhMTEK
  password: YjgwNDc4ZDY4NmQzNzQzYjNkYmUwYmE3YjMwOTM2OWUK
