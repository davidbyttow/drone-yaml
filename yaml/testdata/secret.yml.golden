[
  {
    "kind": "secret",
    "type": "encrypted",
    "data": {
      "password": "YjgwNDc4ZDY4NmQzNzQzYjNkYmUwYmE3YjMwOTM2OWUK",
      "username": "N2NmYjA3ODQwNTY1ODFlY2E5MGJmOWI1NDk0NDFhMTEK"
    }
  }
]